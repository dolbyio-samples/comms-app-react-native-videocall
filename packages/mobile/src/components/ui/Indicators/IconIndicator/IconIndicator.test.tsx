import { defaultTheme as theme } from '@dolbyio/comms-uikit-common';
import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';

import { render, mergeStyles } from '../../../../utils/test-utils';

import IconIndicator from './IconIndicator';

const { colors } = theme;

const icon = 'microphone';
const testID = 'testID';
const backgroundColor = 'primary.500';
const iconColor = 'primary.500';

describe('IconIndicator component', () => {
  test('Passes TestID', () => {
    const { getByTestId } = render(<IconIndicator testID={testID} icon={icon} />);
    expect(getByTestId(testID)).not.toBeNull();
  });
  test('Passes given styles', () => {
    const size = 'small';
    const expectedWidth = 24;
    const { getByTestId } = render(
      <IconIndicator testID={testID} backgroundColor={backgroundColor} icon={icon} size={size} />,
    );
    const element = getByTestId(testID);
    const styles = mergeStyles(element.props.style);
    expect(styles.backgroundColor).toBe(colors.primary[500]);
    expect(styles.width).toBe(expectedWidth);
  });
  test('Uses default size as medium', () => {
    const expectedWidth = 32;
    const expectedHeight = 32;
    const { getByTestId } = render(<IconIndicator testID={testID} backgroundColor={backgroundColor} icon={icon} />);
    const element = getByTestId(testID);
    const styles = mergeStyles(element.props.style);
    expect(styles.backgroundColor).toBe(colors.primary[500]);
    expect(styles.width).toBe(expectedWidth);
    expect(styles.height).toBe(expectedHeight);
  });
  test('Uses default backgroundColor as whiteAlpha', () => {
    const size = 'medium';
    const expectedWidth = 32;
    const { getByTestId } = render(<IconIndicator testID={testID} icon={icon} size={size} />);
    const element = getByTestId(testID);
    const styles = mergeStyles(element.props.style);
    expect(styles.backgroundColor).toBe(colors.whiteAlpha[500]);
    expect(styles.width).toBe(expectedWidth);
  });
  test('Uses given iconColor', () => {
    const size = 'small';
    const expectedWidth = 24;
    const expectedHeight = 24;
    const { getByTestId } = render(<IconIndicator testID={testID} icon={icon} iconColor={iconColor} size={size} />);
    const element = getByTestId(testID);
    const styles = mergeStyles(element.props.style);
    expect(styles.backgroundColor).toBe(colors.whiteAlpha[500]);
    expect(styles.width).toBe(expectedWidth);
    expect(styles.height).toBe(expectedHeight);
  });
});

describe('IconIndicator component snapshot', () => {
  test('matches when optional parameters not given', () => {
    const tree = renderer.create(<IconIndicator testID={testID} icon={icon} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('matches when all parameters given', () => {
    const size = 'small';
    const tree = renderer
      .create(
        <IconIndicator
          testID={testID}
          backgroundColor={backgroundColor}
          icon={icon}
          iconColor={iconColor}
          size={size}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
