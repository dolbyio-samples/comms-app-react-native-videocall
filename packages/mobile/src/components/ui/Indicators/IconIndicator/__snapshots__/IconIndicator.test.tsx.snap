// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconIndicator component snapshot matches when all parameters given 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      Object {
        "height": 24,
        "width": 24,
      },
      Object {
        "backgroundColor": "#8829CC",
      },
    ]
  }
  testID="testID"
>
  <svg
    data-file-name="SvgMicrophone"
    fill="#8829CC"
    height={16}
    width={16}
  />
</View>
`;

exports[`IconIndicator component snapshot matches when optional parameters not given 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 100,
        "height": 32,
        "justifyContent": "center",
        "width": 32,
      },
      false,
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.36)",
      },
    ]
  }
  testID="testID"
>
  <svg
    data-file-name="SvgMicrophone"
    fill="#FFFFFF"
    height={18}
    width={18}
  />
</View>
`;
